Block/Sales/View.php

```diff
@@ -55,49 +35,58 @@ public function _construct()
+        if (! is_array($data)) {
+            return '';
+        }
 
-        if ($this->helper->isPickupLocation($data)) {
-            if (is_array($data) && key_exists('pickupLocation', $data)) {
-                $html .= __($data['carrier'] . ' location:') . ' ' . $dateTime;
-                if ($data['deliveryType'] != 'pickup') {
-                    $html .= ', ' . __($data['deliveryType']);
-                }
-                $html .= ', ' . $data['pickupLocation']['location_name'] . ', ' . $data['pickupLocation']['city'] . ' (' . $data['pickupLocation']['postal_code'] . ')';
-            } else {
-                /** Old data from orders before version 1.6.0 */
-                $html .= __('MyParcel options data not found');
-            }
-        } else {
-            if (is_array($data) && key_exists('date', $data)) {
-                if (key_exists('packageType', $data)) {
-                    $html .= __($data['packageType'] . ' ');
+        $date            = new DateTime($data['date'] ?? '');
+        $dateTime        = $date->format('d-m-Y H:i');
+        $deliveryOptions = DeliveryOptionsAdapterFactory::create((array) $data);
+
+        ob_start();
+
+        if ($deliveryOptions->isPickup()) {
+            try {
+                echo __("{$data['carrier']} location:"), ' ', $dateTime;
+
+                if ($data['deliveryType'] !== 'pickup') {
+                    echo ', ', __($data['deliveryType']);
                 }
 
-                $html .= __('Deliver:') . ' ' . $dateTime;
+                $pickupLocation = $deliveryOptions->getPickupLocation();
 
-                if (key_exists('shipmentOptions', $data)) {
-                    if (key_exists('signature', $data['shipmentOptions']) && $data['shipmentOptions']['signature']) {
-                        $html .= ', ' . __('Signature on receipt');
-                    }
-                    if (key_exists('only_recipient', $data['shipmentOptions']) && $data['shipmentOptions']['only_recipient']) {
-                        $html .= ', ' . __('Home address only');
-                    }
+                if (null !== $pickupLocation) {
+                    echo ', ', $pickupLocation->getLocationName(), ', ', $pickupLocation->getCity(), ' (', $pickupLocation->getPostalCode(), ')';
                 }
+
+            } catch (\Throwable $e) {
+                ObjectManager::getInstance()->get(CheckoutHelper::class)->log($e->getMessage());
+                echo __('MyParcel options data not found');
             }
-        }
+        } elseif (array_key_exists('date', $data)) {
+            if (array_key_exists('packageType', $data)) {
+                echo __($data['packageType']), ' ';
+            }
+
+            echo __('Deliver:'), ' ', $dateTime;
+
+            $shipmentOptions = $deliveryOptions->getShipmentOptions();
 
-        if (is_array($data) && key_exists('browser', $data)) {
-            $html = ' <span title="' . $data['browser'] . '">' . $html . '</span>';
```

Comment:
Deze informatie wordt, volgens mij, in de nieuwe variant niet meer weergeven, kan niet helemaal beoordelen wat het is en of dat erg is, dus ook weer even een sanity check :)

Reply:
Scherp gezien. Deze key heb ik nog nooit gezien, en ook na een zoektocht in onze huidige versies nergens aangetroffen. Deze wordt ook in onze huidige versies nergens ‘geplaatst’, dus ik denk dat het legacy is.